contracts/tasks/deploy.ts

```diff
@@ -1,29 +1,48 @@
 import { task } from "hardhat/config";
+import { writeFile } from "node:fs/promises";
 
-task("deploy", "Deploy contracts and verify").setAction(async ({}, { ethers, upgrades }) => {
-  const HypercertMinter = await ethers.getContractFactory("HypercertMinter");
-  const hypercertMinter = await upgrades.deployProxy(HypercertMinter, {
-    kind: "uups",
-    unsafeAllow: ["constructor"],
-  });
-  await hypercertMinter.deployed();
-  console.log(`HypercertMinter is deployed to proxy address: ${hypercertMinter.address}`);
+task("deploy", "Deploy contracts and verify")
+  .addOptionalParam("output", "write the details of the deployment to this file if this is set")
+  .setAction(async ({ output }, { ethers, upgrades }) => {
+    const HypercertMinter = await ethers.getContractFactory("HypercertMinter");
+    const hypercertMinter = await upgrades.deployProxy(HypercertMinter, {
+      kind: "uups",
+      unsafeAllow: ["constructor"],
+    });
+    await hypercertMinter.deployed();
+    console.log(`HypercertMinter is deployed to proxy address: ${hypercertMinter.address}`);
 
-  if (hre.network.name !== "hardhat" && hre.network.name !== "localhost") {
-    try {
-      const code = await hypercertMinter.instance?.provider.getCode(hypercertMinter.address);
-      if (code === "0x") {
-        console.log(`${hypercertMinter.name} contract deployment has not completed. waiting to verify...`);
-        await hypercertMinter.instance?.deployed();
-      }
-      await hre.run("verify:verify", {
-        address: hypercertMinter.address,
-      });
-    } catch ({ message }) {
-      if ((message as string).includes("Reason: Already Verified")) {
-        console.log("Reason: Already Verified");
+    // If the `deploymentFile` option is set then write the deployed address to
+    // a json object on disk. This is intended to be deliberate with how we
+    // output the contract address and other contract information.
+    if (output) {
+      const contract = await hypercertMinter.deployed();
```

Comment:
Tiny detail: this can alse be move to line 12 since it's the same await

Reply:
oh whoops :) totally missed that 👍